synapse/app/homeserver.py

```diff
@@ -366,6 +366,16 @@ def setup(config_options: List[str]) -> SynapseHomeServer:
                 "`enable_registration_without_verification` config option to `true`."
             )
 
+    if (
+        config.experimental.faster_joins_enabled
+        and config.worker.worker_app is not None
+    ):
```

Comment:
To check: in a worker deployment, the main process has `config.worker.worker_app` set to None?

I'm worried that there might be danger in having the main process run some remote-join-aware code, while others are expecting full-join-only. But maybe there isn't an easy way round this?

Reply:
Yeah, this condition will never be hit I don't think, as `synapse.app.homeserver` is the entry point for master only.

> I'm worried that there might be danger in having the main process run some remote-join-aware code, while others are expecting full-join-only. But maybe there isn't an easy way round this?

I don't think this is an issue as long as we refuse to start up workers, as the only real difference between monolith and worker mode is if there are other workers running.